google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -304,12 +304,15 @@ def run(
                 should be peered. For example, projects/12345/global/networks/myVPC.
 
                 Private services access must already be configured for the network.
-                If left unspecified, the job is not peered with any network.
+                If left unspecified, the network set in aiplatform.init will be used.
+                Otherwise, the Endpoint is not peered with any network.
             sync (bool):
                 Optional. Whether to execute this method synchronously. If False, this method will unblock and it will be executed in a concurrent Future.
             create_request_timeout (float):
                 Optional. The timeout for the create request in seconds.
         """
+        network = network or initializer.global_config.network
```

Comment:
Note that for methods decorated with `optional_sync` and with `sync` parameter the default argument needs to be captured synchronously at call time. Otherwise the argument may change between call time and execution time if the use uses `init(network=...)` again. So something like

```Python

def run( self, ..., network, sync=True):
    network = network of initializer.global_config.network
    return self._run(..., network=network, sync=sync)

@base.optional_sync
def _run(..., network, sync=True):
  ...
```

Reply:
Hey Sasha, I checked through the files that would be affected by this (`jobs.py`, `pipeline_jobs.py`, `training_jobs.py`, `models.py`, `matching_engine_index_endpoint.py`) and they all have explicit passing of the `network` arg into their associated helper methods. So, I think they are good to go

Reply:
Can we look into this some more?  I think Sasha is saying the public method should not be decorated with `optional_sync`, and the public method should call a private method decorated with `optional_sync`.

Reply:
hey @sasha-gitg can we get clarification on this?

Reply:
Yu-han is right. The current implementation, without the proposed change above, could result in thisunintened behavior:

```Python
aiplatform.init(project='my-project')

job = aiplatform.PipelineJob(...)
job.run(..., sync=False)

aipaltform.init(network='my_network') # job is potentially executed with this network which is not the intention
```

Reply:
Fixe, added and moved `@base.optional_sync` to a helper method to capture network arg synchronously (`network = network of initializer.global_config.network`) to `matching_engine_index_endpoint.py`, `pipeline_jobs.py`, and `jobs.py`